operator/internal/manifests/distributor.go

```diff
@@ -134,6 +134,10 @@ func NewDistributorDeployment(opts Options) *appsv1.Deployment {
 
 	if opts.Stack.Replication != nil {
 		podSpec.TopologySpreadConstraints = append(podSpec.TopologySpreadConstraints, topologySpreadConstraints(*opts.Stack.Replication, LabelDistributorComponent, opts.Name)...)
+		if len(opts.Stack.Replication.Zones) > 0 {
+			resetEnvVar(&podSpec, availibilityZoneEnvVarName)
+			podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, getInstanceAvailabilityZoneEnvVar())
```

Comment:
`podSpec` is produced by this function. I think it would be much more straight-forward to produce the environment variable configuration directly in the beginning of the function and use that when creating `podSpec` instead of creating it with empty env variables first and then, possibly, updating the value later.